From 5f4b04d1f2bb898e027e6506618830969b39ad0a Mon Sep 17 00:00:00 2001
From: Ning Sun <ning.sun@intel.com>
Date: Mon, 24 Jul 2017 05:34:17 -0700
Subject: [PATCH 02/22] Reset debug PCR16 to zero. PCR16 is used to identify
 TPM2.0 supported banks and hash algorithms. Upon successful execution,
 TPM2_PCR_Event command will extend a hashed value in each discovered bank of
 PCR16. PCR16 needs to be reset to zero after this discovery.

Signed-off-by: Ning Sun <ning.sun@intel.com>
(cherry picked from commit 7bf8782decfe8f7f9d30b5d1a0690b23c61600bf)
---
 tboot/common/tpm_20.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tboot/common/tpm_20.c b/tboot/common/tpm_20.c
index b9b67c9..e4a22b4 100644
--- a/tboot/common/tpm_20.c
+++ b/tboot/common/tpm_20.c
@@ -2555,6 +2555,12 @@ static bool tpm20_init(struct tpm_if *ti)
     for (unsigned int i=0; i<ti->alg_count; i++)
         printk(TBOOT_INFO"tboot: hash alg = %08X\n", ti->algs[i]);
 
+    /* reset debug PCR 16 */
+    if (!tpm20_pcr_reset(ti, ti->cur_loc, 16)){
+        printk(TBOOT_WARN"TPM: tpm20_pcr_reset failed...\n");
+	return false;
+    }
+
     if (handle2048 != 0)
         goto out;
 
-- 
2.16.1

